src/providers/arcgisrest/qgsafsshareddata.cpp

```diff
@@ -144,39 +144,62 @@ bool QgsAfsSharedData::getFeature( QgsFeatureId id, QgsFeature &f, const QgsRect
     return filterRect.isNull() || ( f.hasGeometry() && f.geometry().intersects( filterRect ) );
   }
 
-  // Fetch 100 features at the time
-  const int startId = ( id / 100 ) * 100;
-  const int stopId = std::min< size_t >( startId + 100, mObjectIds.length() );
+  bool featureFetched = false;
+  int startId, stopId;
   QList<quint32> objectIds;
-  objectIds.reserve( stopId );
-  for ( int i = startId; i < stopId; ++i )
-  {
-    if ( i >= 0 && i < mObjectIds.count() && !mDeletedFeatureIds.contains( i ) && !mCache.contains( i ) )
-      objectIds.append( mObjectIds.at( i ) );
-  }
+  QVariantMap queryData;
```

Comment:
```suggestion
```

Reply:
This has to be declared outside of the while() loop as it's used subsequently (i.e. `const QVariantList featuresData = queryData[QStringLiteral( "features" )].toList();`)

Reply:
ok, fair enough!